src/pages/search/index.tsx

```diff
@@ -139,13 +103,66 @@ function getIsOpenNow(times: Time[]): boolean {
   return isOpenNow;
 }
 
+type OpeningHoursInfoBlockProps = {
+  openingHours: UnifiedSearchOpeningHours;
+  openAt: Date | undefined;
+  locale: Locale;
+};
+
+function OpeningHoursInfoBlock({
+  openingHours,
+  openAt,
+  locale,
+}: OpeningHoursInfoBlockProps): React.ReactElement<
+  React.ComponentProps<typeof InfoBlock>
+> | null {
+  const { t } = useTranslation("search_page");
+
+  const openingHourTimes =
+    (openAt
+      ? openingHours?.data?.find(
+          (timeData) => timeData.date === format(openAt, "yyyy-MM-dd")
+        )?.times
+      : openingHours?.today) ?? [];
+  const isOpenNow = getIsOpenNow(openingHourTimes);
+  const openAtFilterIsToday = isToday(openAt);
+  const humanizedOpeningHours = humanizeOpeningHour(
+    {
+      date: new Date().toJSON(),
+      times: openingHourTimes as Time[],
+    },
+    locale,
+    "short"
+  );
+
+  if (!humanizedOpeningHours) {
+    return null;
+  }
+
+  return (
+    <InfoBlock
+      target="card"
+      headingLevel="h3"
+      icon={<IconClock aria-hidden="true" />}
+      name={
+        !(openAt && openAtFilterIsToday) && isOpenNow
```

Comment:
Rather complex condition inlined here. At least I can't easily understand this 🤔

Reply:
Is it possible for `isOpenNow` to be `true` but we wouldn't render open now label?

Reply:
I spent some time with this, but I couldn't find a clearer way.

I'll make a helper function so that we can multiline it and maybe make it easier to read that way.

When an `openAt` filter is selected, we want to use that date even if the place would be ope now.